src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { currentTodoSlice, isSelected } from '../../features/currentTodo';
+import { useAppDispatch, useAppSelector } from '../../app/store';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const todoModal = useAppSelector(isSelected);
+  const dispatch = useAppDispatch();
+
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (todoModal) {
+      setIsLoading(true);
+      getUser(todoModal.userId)
+        .then(resp => setUser(resp))
+        .finally(() => setIsLoading(false));
```

Comment:
Consider adding error handling for the `getUser` API call. This will help manage any potential errors that occur during the API request and improve the user experience.